feat(aio): enable deep-linking on deployed apps (until prerendering is done)

gkalpak opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
Deep-linking does not work on `ngbuilds.io` (PR previews) or `aio-stading.firebaseapp.com` (staging).
This was intentional, since the ultimate goal is to prerender all pages. Until this isimplemented, though, it is useful to support deep-linking.

**What is the new behavior?**
Deep-linking works on `ngbuilds.io` (PR previews) or `aio-stading.firebaseapp.com` (staging).


**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

Other information:
Partly addresses #14922.

gkalpak committed:
feat(aio): enable deep-linking on deployed apps (until prerendering is done)

petebacondarwin commented on aio/aio-builds-setup/dockerbuild/nginx/aio-builds.conf:
```diff
@@ -14,9 +14,13 @@ server {
   index            index.html;
 
-  location / {
+  location "~/[^/]+\.[^/]+$" {
```
what does the `~` char mean here?

gkalpak replied:
`~` means it is a RegExp.

gkalpak commented:
Squashed the commits and also added a [small fix][1] for the random test failures (the tests might or might not fail based on their relative order).

<sub>(I had run into the same problem [before][2], so I should have known better :smiley:)</sub> 

[1]: https://github.com/angular/angular/pull/15049/files#diff-3b5c3eb4d712b74cde66f32ebf5ab531L197
[2]: https://github.com/gkalpak/angular/blob/69b8836756ea3c332cb9cde1116862bb04075c8b/aio/aio-builds-setup/dockerbuild/scripts-js/test/upload-server/build-creator.spec.ts#L57-L63

gkalpak commented:
@mary-poppins, if you think this is ready for merging, please deploy a preview and post the link here.

mary-poppins commented:
The angular.io preview for 69b8836 is available [here][1].

[1]: https://pr15049-69b8836756ea3c332cb9cde1116862bb04075c8b.ngbuilds.io/

gkalpak commented:
It worked! :tada:

mary-poppins commented:
The angular.io preview for 81253a4 is available [here][1].

[1]: https://pr15049-81253a483965c3b6f51142235861cac9673947b3.ngbuilds.io/
